models/articles.model.js

```diff
@@ -1,10 +1,24 @@
 const db = require("../db/connection.js")
 
-exports.selectArticles = () => {
-    return db.query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;`)
-    .then((result) => {
-        return result.rows
-    })
+exports.selectArticles = (topicQuery) => {
+    if(topicQuery){
+        return db.query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id WHERE articles.topic = $1 GROUP BY articles.article_id ORDER BY created_at DESC;`, [topicQuery])
+        .then((result) => {
+            if(result.rows.length === 0){
+                return Promise.reject({
+                    status: 404,
+                    msg: 'Topic does not exist'
+                })
+            }
+            return result.rows
+        })
+    }
+    else {
+        return db.query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;`)
+        .then((result) => {
+            return result.rows
+        })
+    }
```

Comment:
There is a lot of duplicate code here, can you think of a way to refactor this into using just one query with it being built it up dynamically?